README.md

```diff
@@ -46,57 +61,117 @@ api("com.github.open-feature:kotlin-sdk:[ANY_BRANCH]-SNAPSHOT")
 
 This will get a build from the head of the mentioned branch. 
 
-### Maven
+#### Maven
 
 Installation via Maven Central is currently [WIP](https://github.com/open-feature/kotlin-sdk/issues/37)
 
+### Usage
+
+```kotlin
+// configure a provider and get client
+OpenFeatureAPI.setProvider(customProvider)
+val client = OpenFeatureAPI.getClient()
+
+// get a bool flag value
+client.getBooleanValue("boolFlag", default = false)
+
+// get a bool flag after "ready" signal from provider
+coroutineScope.launch {
+    WithContext(Dispatchers.IO) {
+        client.awaitProviderReady()
+    }
+    client.getBooleanValue("boolFlag", default = false)
+}
+```
+
 ## 🌟 Features
 
-- support for various backend [providers](https://openfeature.dev/docs/reference/concepts/provider)
-- easy integration and extension via [hooks](https://openfeature.dev/docs/reference/concepts/hooks)
-- bool, string, numeric, and object flag types
-- [context-aware](https://openfeature.dev/docs/reference/concepts/evaluation-context) evaluation
+<!-- TODO: update table to indicate implemented features (see legend below) -->
+
+| Status | Features                        | Description                                                                                                                        |
+| ------ | ------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| ✅      | [Providers](#providers)         | Integrate with a commercial, open source, or in-house feature management tool.                                                     |
+| ✅      | [Targeting](#targeting)         | Contextually-aware flag evaluation using [evaluation context](https://openfeature.dev/docs/reference/concepts/evaluation-context). |
+| ✅      | [Hooks](#hooks)                 | Add functionality to various stages of the flag evaluation life-cycle.                                                             |
+| ❌      | Logging                         | Integrate with popular logging packages.                                                                                           |
+| ❌      | Named clients                   | Utilize multiple providers in a single application.                                                                                |
+| ⚠️      | [Eventing](#eventing)           | React to state changes in the provider or flag management system.                                                                  |
+| ✅      | [Shutdown](#shutdown)           | Gracefully clean up a provider during application shutdown.                                                                        |
+| ❌      | Extending                       | Extend OpenFeature with custom providers and hooks.                                                                                |
```

Comment:
It looks like this is at least partially done.

Reply:
Could you elaborate on the difference between `Extending` and `Providers` (assuming we ignore the Hooks part of `Extending`)?

Reply:
It's intended to be a place that shows how a basic provider and hook can be implemented. Since hooks aren't supported, you can skip that section.